### ISSUE TITLE
Can't Save Word Document

### ISSUE THREAD

**Issue by shlatchz** [dokan-dev/dokan-dotnet#27]
Hi,

I created a new word file on the Dokan drive, changed its name, opened it, wrote some text and clicked save.
When I clicked save, the word app crashed.
I tested this both in my app and in the Mirror app and the same thing happened.

I'll mention that this operation worked on prev version (0.7.4) and I suspect it has something to do with the EnumerateNamedStreams function.

(Tested on Win8)

Please help :\

**UPDATE:** It works on Win7.

**Comment by Liryna** [comment]
Hi,

Could you retry the mirror with my last commit f49169d6f2f02cbf3064a78d3649d44f88da7700
I do not have word for now :cry: 

And as usual, some logs would be helpful :wink:

**Comment by shlatchz** [comment]
It does work with mirror.exe
How do we make it work with DokanNet?
What logs do you need?

Thanks

**Comment by Liryna** [comment]
https://github.com/dokan-dev/dokany/wiki/How-to-Debug-Dokan
At first, you can compare the logs from C# mirror and C mirror,
Find what is the difference in the sys calls.

On my side, I know there is a difference in this two mirror but do not have the time to look into it for now.

So if you find something, I would be very interested in it .

**Comment by shlatchz** [comment]
One difference is that mirror.exe actually implements the EnumerateNamedStreams function.
Office does use those streams and requests their data:

EnumerateNamedStreams E:\a.docx
        Stream #0 requested.
         Stream ::$DATA
EnumerateNamedStreams E:\a.docx
        Stream #1 requested.
        No more stream.

**Comment by Liryna** [comment]
@shlatchz $DATA stream is the main stream.
But you can try to fake the implementation of EnumerateNamedStreams and answer only $DATA.

**Comment by shlatchz** [comment]
I tried the following:
```
streamSize = 0;
int streamId = Marshal.ReadInt32(enumContext);
if (streamId != 0)
{
    streamName = String.Empty;
    return Trace("EnumerateNamedStreams", fileName, info, NtStatus.NotImplemented, streamId.ToString(), "out " + streamName, "out " + streamSize.ToString());
}
else
{
    streamName = "::$DATA";
    Marshal.WriteInt32(enumContext, ++streamId);
    return Trace("EnumerateNamedStreams", fileName, info, NtStatus.Success, streamId.ToString(), "out " + streamName, "out " + streamSize.ToString());
}
```
and it did help **my** application. Now, word does save the file, but after the saving process, it shows an error msg:

![meep](https://cloud.githubusercontent.com/assets/4594463/10381004/2b49a99e-6e20-11e5-92c4-de1c2ab63eb8.png)

and in the following saves, it doesn't show an error anymore.

When I ran the changed code in the **Mirror** project, it failed during the save process but didn't crash Word's app.

**Comment by Liryna** [comment]
Well done ! Glad to hear it !
It would be nice if you could contribute to the Mirror and add the support :heart:

**Comment by Liryna** [comment]
@shlatchz , could you try your test again with the last version and specially with the last commit https://github.com/dokan-dev/dokan-dotnet/commit/5e64315c7188c2b7c40e8beb5485e60b6b2125aa ?
There is a little chance that it have fixed it

**Comment by viciousviper** [comment]
@shlatchz Any news on this issue?

**Comment by Liryna** [comment]
Since no news.

**Comment by shlatchz** [comment]
I still can't save both doc and docx on the mirror drive.
I managed to save the docx only once, when I initially tried to save a new docx file on the drive. After I clicked save, it popped the save dialog again (because it failed) and then I tried saving again and it did.
After that, when I change the file and click save, it fails and I cant save it again.
Saving a doc file, alerts "The save failed due to out of memory or disk space"
